index.js

```diff
@@ -84,19 +84,29 @@ export class UpdateAPK {
     // You might check {totalSpace, freeSpace} = await RNFS.getFSInfo() to make sure there is room
     const downloadDestPath = `${RNFS.CachesDirectoryPath}/NewApp.apk`;
 
-    const ret = RNFS.downloadFile({
-      fromUrl: remote.apkUrl,
-      toFile: downloadDestPath,
-      begin,
-      progress,
-      background: true,
-      progressDivider
-    });
+    let options = this.options.apkOptions ? this.options.apkOptions : {};
+
+    const ret = RNFS.downloadFile(
+      Object.assign(
+        {
+          fromUrl: remote.apkUrl,
+          toFile: downloadDestPath,
+          begin,
+          progress,
+          background: true,
+          progressDivider
+        },
+        options
+      )
+    );
 
     jobId = ret.jobId;
 
     ret.promise
       .then(res => {
+        if (res['statusCode'] >= 400 && res['statusCode'] <= 599){
```

Comment:
Nice - not really related to the current PR but this repository does not see a lot of PRs so I'm happy to merge this in along with the options change, it's a positive addition I think